tests/unit/test_clip.py

```diff
@@ -1,4 +1,90 @@
+import operator
+import copy
+
+import numpy as np
+
+from jina import Flow, Document, DocumentArray
 from jina.executors import BaseExecutor
+from jinahub.encoder.clip_image import CLIPImageEncoder
+
+
+def test_clip_any_image_shape():
+    encoder = CLIPImageEncoder()
+    docs = DocumentArray([Document(blob=np.ones((224, 224, 3), dtype=np.uint8))])
+
+    encoder.encode(docs=docs, parameters={})
+    assert len(docs.get_attributes('embedding')) == 1
+
+    docs = DocumentArray([Document(blob=np.ones((100, 100, 3), dtype=np.uint8))])
+    encoder.encode(docs=docs, parameters={})
+    assert len(docs.get_attributes('embedding')) == 1
+
+
+def test_clip_batch():
+    encoder = CLIPImageEncoder(default_batch_size=10, model_name='ViT-B/32', device='cpu')
+    docs = DocumentArray([Document(blob=np.ones((224, 224, 3), dtype=np.uint8)) for _ in range(25)])
+    encoder.encode(docs, parameters={})
+    assert len(docs.get_attributes('embedding')) == 25
+
+
+def test_traversal_paths():
+    blob = np.ones((224, 224, 3), dtype=np.uint8)
+    docs = DocumentArray([Document(
+        id='root1',
+        blob=blob,
+        chunks=[
+            Document(
+                id='chunk11',
+                blob=blob,
+                chunks=[
+                    Document(id='chunk111', blob=blob),
+                    Document(id='chunk112', blob=blob),
+                ]
+            ),
+            Document(id='chunk12', blob=blob),
+            Document(id='chunk13', blob=blob),
+        ]
+    )])
+
+    encoder = CLIPImageEncoder(default_traversal_paths = ['c'], model_name = 'ViT-B/32', device = 'cpu')
+    encoder.encode(docs, parameters={})
+    for path, count in [['r', 0], ['c', 3], ['cc', 0]]:
+        assert len(docs.traverse_flat([path]).get_attributes('embedding')) == count
+
+    #results = f.post(on='/test', inputs=docs, parameters={'traversal_paths': ['cc']}, return_results=True)
+    encoder = CLIPImageEncoder(default_traversal_paths=['cc'])
+    encoder.encode(docs, parameters={})
+    for path, count in [['r', 0], ['c', 0], ['cc', 2]]:
+        assert len(docs.traverse_flat([path]).get_attributes('embedding')) == count
+
+
+def test_custom_processing():
+    f = Flow().add(uses=CLIPImageEncoder)
+    with f:
+        result1 = f.post(on='/test', inputs=[Document(blob=np.ones((224, 224, 3), dtype=np.uint8))],
+                         return_results=True)
+
+    f = Flow().add(uses={
```

Comment:
no Flow should be involved in `unittests`

Reply:
fixed